### PR TITLE
Added test case for `internal/errorgroup` 

### PR DESCRIPTION
**Pull request by hlts2**
Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

### Description:

I added a test case for internal/errorgroup.

### Environment:

- Go Version: 1.14.3
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.11.5

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/errgroup/group_test.go [hunk 1]
```diff
@@ -419,194 +423,205 @@ func Test_group_Limitation(t *testing.T) {
...
-				test.beforeFunc(test.args)
-			}
+			defer goleak.VerifyNone(tt)
 			if test.afterFunc != nil {
 				defer test.afterFunc(test.args)
 			}
 			if test.checkFunc == nil {
 				test.checkFunc = defaultCheckFunc
 			}
-			g := &group{
```
Comment (kevindiu) [inline internal/errgroup/group_test.go:514]: should we keep using this part?

File: internal/errgroup/group_test.go [hunk 2]
```diff
@@ -616,21 +631,13 @@ func Test_group_doCancel(t *testing.T) {
...
-				wg:               test.fields.wg,
-				limitation:       test.fields.limitation,
-				enableLimitation: test.fields.enableLimitation,
-				cancelOnce:       test.fields.cancelOnce,
-				mu:               test.fields.mu,
-				emap:             test.fields.emap,
-				errs:             test.fields.errs,
-				err:              test.fields.err,
-			}
+			gi, _ := New(test.args.ctx)
```
Comment (kevindiu) [inline internal/errgroup/group_test.go:634]: same

Comment (hlts2) [reply internal/errgroup/group_test.go:634]: we should use this part.

Comment (kevindiu) [reply internal/errgroup/group_test.go:634]: why??

File: internal/errgroup/group_test.go [hunk 3]
```diff
@@ -187,42 +188,63 @@ func TestGet(t *testing.T) {
...
+		if got, want := got.(*group), w.want.(*group); !reflect.DeepEqual(got.egctx, want.egctx) {
 			return errors.Errorf("got = %v, want %v", got, w.want)
 		}
 		return nil
 	}
+	defaultBeforeFunc := func() {
+		instance, once = nil, sync.Once{}
+	}
+	defaultAfterFunc := func() {
+		defaultBeforeFunc()
```
Comment (kpango) [inline internal/errgroup/group_test.go:200]: dirty code
```suggestion
		defaultBeforeFunc()
```

File: internal/errgroup/group_test.go [hunk 4]
```diff
@@ -307,86 +331,74 @@ func Test_group_Limitation(t *testing.T) {
...
-		       }
-		   }(),
-		*/
+		{
+			name: "set disable when limit is 0",
+			args: args{
+				limit: 0,
+			},
+			fields: fields{
+				limitation: make(chan struct{}),
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/errgroup/group_test.go:382]: maybe we can delete the limitation field here?

[the remark this answers is not in the document]
Comment (hlts2) [reply internal/errgroup/group_test.go:382]: sorry. I understand your opinion:bow:
I will fix it soon.

File: internal/errgroup/group_test.go [hunk 5]
```diff
@@ -307,86 +331,71 @@ func Test_group_Limitation(t *testing.T) {
...
+			},
+		},
+
+		{
+			name: "set enable when limit is 1",
+			args: args{
+				limit: 1,
+			},
+			fields: fields{
+				limitation: make(chan struct{}),
```
Comment (kevindiu) [inline internal/errgroup/group_test.go:382]:
```suggestion
				limitation: make(chan struct{}, 1),
```

File: internal/errgroup/group_test.go [hunk 6]
```diff
@@ -729,81 +702,75 @@ func Test_group_Wait(t *testing.T) {
...
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		func() test {
+			var num int32
+			return test{
+				name: "returns nil after all goroutne returns",
+				fields: fields{
+					limitation: make(chan struct{}),
```
Comment (kevindiu) [inline internal/errgroup/group_test.go:710]:
```suggestion
```

File: internal/errgroup/group_test.go [hunk 7]
```diff
@@ -729,81 +702,75 @@ func Test_group_Wait(t *testing.T) {
...
+					}
+					return nil
+				},
+			}
+		}(),
+
+		{
+			name: "returns error when g.errs is not nil",
+			fields: fields{
+				limitation: make(chan struct{}),
```
Comment (kevindiu) [inline internal/errgroup/group_test.go:739]:
```suggestion
```

**Pull request merged**